File: files/files/2009/04/05/oo.php
<?php

class Foo {
    public function __construct($name) {
        $this->name = $name;
    }

    public function emote() {
        print $this->name . " is emoting.\n";
    }
}

// Construct the objects
$o1 = new Foo("Foo one");
$o2 = new Foo("Foo two");

// Call the methods directly
$o1->emote();
$o2->emote();

// Call the methods by reference
$f = $o1->emote;

File: files/files/2009/04/05/oo.py
#!/usr/bin/env python
"""
This is a simple programme which demonstrates the use of references to methods
on objects in Python. This can be constrasted with the accompanying files
demonstrating similar techniques in Javascript, PHP, and Ruby.

For more information see the articles discussing this issue at:
    <http://passingcuriosity.com/2009/object-oriented-function-references>
"""

class Foo:
    def __init__(self, name):
        self.name = name

    def emote(self):
        print "%s is emoting" % self.name

if __name__ == "__main__":
    # Call the method directly 
    o1 = Foo("foo one")
    o2 = Foo("foo two")
    o1.emote()
    o2.emote()

    # Call the method by reference
    f = o1.emote
    f()
    f = o2.emote
    f()
